internal/lm/imex.go

```diff
@@ -17,92 +17,26 @@
 package lm
 
 import (
-	"bufio"
-	"errors"
 	"fmt"
-	"io"
-	"net"
-	"os"
-	"path/filepath"
-	"sort"
 	"strings"
 
-	"github.com/google/uuid"
 	"k8s.io/klog/v2"
 
 	spec "github.com/NVIDIA/k8s-device-plugin/api/config/v1"
 	"github.com/NVIDIA/k8s-device-plugin/internal/resource"
 )
 
-const (
-	// ImexNodesConfigFilePath is the path to the IMEX nodes config file.
-	// This file contains a list of IP addresses of the nodes in the IMEX domain.
-	ImexNodesConfigFilePath = "/etc/nvidia-imex/nodes_config.cfg"
-)
-
 func newImexLabeler(config *spec.Config, devices []resource.Device) (Labeler, error) {
-	var errs error
-	for _, root := range imexNodesConfigFilePathSearchRoots(config) {
-		configFilePath := filepath.Join(root, ImexNodesConfigFilePath)
-		imexLabeler, err := imexLabelerForConfigFile(configFilePath, devices)
-		if err != nil {
-			errs = errors.Join(errs, err)
-			continue
-		}
-		if imexLabeler != nil {
-			klog.Infof("Using labeler for IMEX config %v", configFilePath)
-			return imexLabeler, nil
-		}
-	}
-	if errs != nil {
-		return nil, errs
-	}
-
-	return empty{}, nil
-}
-
-// imexNodesConfigFilePathSearchRoots returns a list of roots to search for the IMEX nodes config file.
-func imexNodesConfigFilePathSearchRoots(config *spec.Config) []string {
-	// By default, search / and /config for config files.
-	roots := []string{"/", "/config"}
-
-	if config == nil || config.Flags.Plugin == nil || config.Flags.Plugin.ContainerDriverRoot == nil {
-		return roots
-	}
-
-	// If a driver root is specified, it is also searched.
-	return append(roots, *config.Flags.Plugin.ContainerDriverRoot)
-}
-
-func imexLabelerForConfigFile(configFilePath string, devices []resource.Device) (Labeler, error) {
-	imexConfigFile, err := os.Open(configFilePath)
-	if os.IsNotExist(err) {
-		// No imex config file, return empty labels
-		return nil, nil
-	} else if err != nil {
-		return nil, fmt.Errorf("failed to open imex config file: %v", err)
-	}
-	defer imexConfigFile.Close()
-
 	clusterUUID, cliqueID, err := getFabricIDs(devices)
 	if err != nil {
 		return nil, err
```

Comment:
```suggestion
		return empty{}, nil
```

Reply:
No. Here we are returning an actual error. Do you have a motivation as to why we want to ignore it?